src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,7 +153,39 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            var allRenameLocations = BaseViewModel.Session.AllRenameLocationsTask.Join();
+            ImmutableDictionary<string, string[]> context = default;
+            if (allRenameLocations.Locations.Count > 0)
+            {
+                var references = new List<string>();
+                foreach (var renameLocation in allRenameLocations.Locations)
+                {
+                    var syntaxTree = renameLocation.Document.GetSyntaxTreeSynchronously(_cancellationTokenSource.Token);
+                    var text = syntaxTree.GetText(_cancellationTokenSource.Token);
```

Comment:
same issue.  also very expensive.  should be Document.GetTextAsync.

Reply:
done!